### DISCUSSION TITLE
Does Pactum support JsonPath assertion?

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#192] [Q&A] [answered]
When you have large Json with huge list of similar arrays in that json, verifying a value with reference from another data will help.
Using Jsonpath (please refer "https://goessner.net/articles/JsonPath/ ")  we can verify it easily.
Does pactum support Jsonpath ?

If not then , is ther any way to assert data with reference to another data in a json ?

**Comment by ASaiAnudeep** [comment, marked as answer]
Almost all assertion methods support **JsonPath** provided by [json-query](https://www.npmjs.com/package/json-query). Please refer the [docs](https://pactumjs.github.io/api/assertions/expectJson.html)

**Reply by crystalfp** [reply]
Thanks! But how to test a JSON don't have a certain field? I tried with:
```
				.expectJsonMatch({
					header: {
						note: "$V === undefined"
					}
				})
```
But it complains that `Json doesn't have property 'note' at '$.header'`.
The workaround is to use jest:
```
				.expect((ctx) => {
					expect(ctx.res.json.header).not.toContainKey("note");
				})
```
But seems to me I'm going back to jest too much.
Could some form of expectNotSomething() to be added to Pactum?
Thanks for clarifying!
mario

**Reply by ASaiAnudeep** [reply]
It's easy to use [notIncludes](https://pactumjs.github.io/api/matching/notIncludes.html) to assert such conditions.

```js
const { spec } = require('pactum');
const { notIncludes } = require('pactum-matchers');

await spec()
  .get('<url>')
  .expectJsonMatch({
    "header": notIncludes("note")
  });
```

**Comment by mihirsasmal** [comment]
Thanks, ASaiAnudeep for the reply.

I was using all the assertions mentioned in docs but trying to use Jsonpath usages based on this "https://goessner.net/articles/JsonPath/ " that's why it wasn't working for me.

After exploring [json-query](https://www.npmjs.com/package/json-query) based Jsonpath from your suggestion and docs , I believe its the easier and best way to find jsonpath than using the xpath like syntax from "https://goessner.net/articles/JsonPath/ " and using any online json path finder.

Thanks
Mihir
